File: streamx-flink/streamx-flink-shims/streamx-flink-shims_flink-1.13/src/main/scala/com/streamxhub/streamx/flink/core/TableContext.scala
/*
 * Copyright (c) 2019 The StreamX Project
 * <p>
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements. See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership. The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at
 * <p>
 * http://www.apache.org/licenses/LICENSE-2.0
 * <p>
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
package com.streamxhub.streamx.flink.core

import org.apache.flink.api.java.utils.ParameterTool
import org.apache.flink.table.api.TableEnvironment
import org.apache.flink.table.descriptors.{ConnectTableDescriptor, ConnectorDescriptor}
import org.apache.flink.table.module.ModuleEntry


class TableContext(override val parameter: ParameterTool,
                   private val tableEnv: TableEnvironment) extends FlinkTableTrait(parameter, tableEnv) {

  /**
   * for scala
   *
   * @param args
   */
  def this(args: (ParameterTool, TableEnvironment)) = this(args._1, args._2)

  /**
   * for java
   *
   * @param args
   */
  def this(args: TableEnvConfig) = this(FlinkTableInitializer.initJavaTable(args))


  override def useModules(strings: String*): Unit = tableEnv.useModules(strings: _*)


  override def listFullModules(): Array[ModuleEntry] = tableEnv.listFullModules()

  @Deprecated override def connect(connectorDescriptor: ConnectorDescriptor): ConnectTableDescriptor = tableEnv.connect(connectorDescriptor)

}

File: streampark-flink/streampark-flink-shims/streampark-flink-shims_flink-1.19/src/main/java/org/apache/streampark/flink/core/TableContext.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.apache.streampark.flink.core;

import org.apache.streampark.flink.configuration.FlinkJobParameters;
import org.apache.streampark.flink.core.bean.TableContextSpec;

import org.apache.flink.table.api.TableEnvironment;

/** Flink 1.19 table environment context. */
public class TableContext extends AbstractFlinkTable {

    public TableContext(FlinkJobParameters parameter, TableEnvironment tableEnv) {
        super(parameter, tableEnv);
    }

    public TableContext(TableContextSpec contextConfig) {
        this(contextConfig.parameter, contextConfig.tableEnv);
    }

}
